pkg/errcode/errcode.go

```diff
@@ -82,6 +82,66 @@ const (
 	ErrNoMatchingServiceForServiceAccount
 )
 
+// Range 4000-3050 reserved for errors related to certificate providers
+const (
+	// ErrFetchingCertSecret indicates a secret containing a certificate could not be fetched
+	ErrFetchingCertSecret ErrCode = iota + 4000
+
+	// ErrObtainingCertFrom indicates a certificate could not be obtained from a secret
+	ErrObtainingCertFromSecret
+
+	// ErrObtainingPrivateKeyFromSecret indicates the certificate's private key could not be obtained from a secret
+	ErrObtainingPrivateKeyFromSecret
+
+	// ErrObtainingCertExpirationFromSecret indicates the certificate's expiration could not be obtained from a secret
+	ErrObtainingCertExpirationFromSecret
+
+	// ErrParsingCertExpiration indicates the certificate's expiration could not be parsed
+	ErrParsingCertExpiration
+
+	// ErrCreatingCertSecret indicates the secret to containing a certificate could not be created
+	ErrCreatingCertSecret
+
+	// ErrGeneratingPrivateKey indicates a private key could not be generated
+	ErrGeneratingPrivateKey
+
+	// ErrEncodingKeyDERtoPEM indicates a private key could not be incoded from DER to PEM
+	ErrEncodingKeyDERtoPEM
+
+	// ErrCreatingCertReq indicates a certificate request could not be created
+	ErrCreatingCertReq
+
+	// ErrEncodingCertRqDERtoPEM indicates a certificate request could not be encodef from DER to PEM
+	ErrEncodingCertReqDERtoPEM
```

Comment:
This shouldn't be different than ErrEncodingCertDERToPEM, let's reuse the same because only the function names are different prior to the same errors.